proto/uber/cadence/api/v1/service_workflow.proto

```diff
@@ -108,6 +111,17 @@ message RestartWorkflowExecutionRequest {
   string reason = 4;
 }
 
+message DiagnoseWorkflowExecutionRequest {
+  string domain = 1;
+  WorkflowExecution workflow_execution = 2;
+  string identity = 3;
+}
+
+message DiagnoseWorkflowExecutionResponse {
+  string domain = 1;
+  WorkflowExecution workflow_execution = 2;
```

Comment:
nit a better name to be explicit on what is in the response workfow.
```suggestion
  WorkflowExecution diagnostic_workflow_execution = 2;
```

Reply:
updated